rust/worker/src/execution/operators/group.rs

```diff
@@ -0,0 +1,223 @@
+use crate::errors::{ChromaError, ErrorCodes};
+use crate::execution::data::data_chunk::DataChunk;
+use crate::execution::operator::Operator;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub struct GroupOperator {
+    // Approximatedly the number of records in a group. Since we are trying to
```

Comment:
Why is this approximate - I don't follow?

Reply:
Oh I see if the N records for a key > group_size we spill. Can we not say approximate? That is confusing. Can we just call out the case where it may spill. I.E This is a "Soft limit"